docs/source/user-guide/sql/scalar_functions.md

```diff
@@ -3346,6 +3347,30 @@ trim_array(array, n)
   Can be a constant, column, or function, and any combination of array operators.
 - **n**: Element to trim the array.
 
+### `unnest`
+
+Transforms an array into rows.
+
+#### Arguments
+
+- **array**: Array expression to unnest.
+  Can be a constant, column, or function, and any combination of array operators.
+
+#### Example
+
+```
+> select unnest(make_array(1, 2, 3, 4, 5));
++------------------------------------------------------------------+
+| unnest(make_array(Int64(1),Int64(2),Int64(3),Int64(4),Int64(5))) |
++------------------------------------------------------------------+
+| 1                                                                |
+| 2                                                                |
+| 3                                                                |
+| 4                                                                |
+| 5                                                                |
++------------------------------------------------------------------+
+```
+
```

Comment:
unnest also works for structs. Can you possibly add this example too?

```sql
> create table foo as values (named_struct('a', 5, 'b', 'a string'));
0 row(s) fetched.
Elapsed 0.008 seconds.

> select * from foo;
+---------------------+
| column1             |
+---------------------+
| {a: 5, b: a string} |
+---------------------+
1 row(s) fetched.
Elapsed 0.003 seconds.

> select unnest(column1) from foo;
+-----------------------+-----------------------+
| unnest(foo.column1).a | unnest(foo.column1).b |
+-----------------------+-----------------------+
| 5                     | a string              |
+-----------------------+-----------------------+
1 row(s) fetched.
Elapsed 0.003 seconds.
```

Reply:
We could merge this PR as is, as well

Reply:
Yeah, sure. This should be in `Struct Functions` right?

Reply:
I would also love to add a range example
```
select unnest(range(0, 100000))
```

Reply:
> Yeah, sure. This should be in Struct Functions right?

I think that would make sense.


FWIW here is the example:

```sql
DataFusion CLI v39.0.0
> select unnest(range(0, 10));
+-----------------------------------+
| unnest(range(Int64(0),Int64(10))) |
+-----------------------------------+
| 0                                 |
| 1                                 |
| 2                                 |
| 3                                 |
| 4                                 |
| 5                                 |
| 6                                 |
| 7                                 |
| 8                                 |
| 9                                 |
+-----------------------------------+
10 row(s) fetched.
Elapsed 0.054 seconds.
```

Reply:
Added docs for struct unnest and example this range for array.
Couldn't link to it because of the name.
Is there a way to assign custom id to a block with sphinx? Tried labels, didn't work, so named it `unnest (struct)`.

Reply:
> Is there a way to assign custom id to a block with sphinx? Tried labels, didn't work, so named it unnest (struct).

I am not sure to be honest